Public/Src/Engine/Scheduler/Graph/GraphFragmentBuilderTopSort.cs

```diff
@@ -29,12 +32,48 @@ public class GraphFragmentBuilderTopSort : GraphFragmentBuilder
         private readonly ConcurrentDictionary<PipId, IList<Pip>> m_pipDependents = new ConcurrentDictionary<PipId, IList<Pip>>();
         private readonly ConcurrentBigMap<FileArtifact, PipId> m_fileProducers = new ConcurrentBigMap<FileArtifact, PipId>();
         private readonly ConcurrentBigMap<DirectoryArtifact, PipId> m_opaqueDirectoryProducers = new ConcurrentBigMap<DirectoryArtifact, PipId>();
+        private readonly PipStaticFingerprinter m_pipStaticFingerprinter;
+        private readonly PipGraphStaticFingerprints m_pipStaticFingerprints = new PipGraphStaticFingerprints();
 
         /// <summary>
         /// Creates an instance of <see cref="GraphFragmentBuilder"/>.
         /// </summary>
         public GraphFragmentBuilderTopSort(LoggingContext loggingContext, PipExecutionContext pipExecutionContext, IConfiguration configuration) : base(loggingContext, pipExecutionContext, configuration)
         {
+            var extraFingerprintSalts = new ExtraFingerprintSalts(
+                configuration,
+                PipFingerprintingVersion.TwoPhaseV2,
+                configuration.Cache.CacheSalt,
+                new DirectoryMembershipFingerprinterRuleSet(configuration, pipExecutionContext.StringTable).ComputeSearchPathToolsHash());
+
+            m_pipStaticFingerprinter = new PipStaticFingerprinter(
+                pipExecutionContext.PathTable,
+                GetSealDirectoryFingerprint,
+                GetDirectoryProducerFingerprint,
+                extraFingerprintSalts,
+                PathExpander.Default)

```

Comment:
>PathExpander.Default [](start = 16, length = 20)

I think you need a real mount path expander here. System mounts get tokenized with the real mount path expander.